flutter/lib/src/integrations/native_app_start_integration.dart

```diff
@@ -21,18 +53,74 @@ class NativeAppStartIntegration extends Integration<SentryFlutterOptions> {
         options.logger(SentryLevel.debug,
             'Scheduler binding is null. Can\'t auto detect app start time.');
       } else {
-        schedulerBinding.addPostFrameCallback((timeStamp) {
+        schedulerBinding.addPostFrameCallback((timeStamp) async {
+          if (_native.didFetchAppStart) {
+            setAppStartInfo(null);
+            return;
+          }
+
           // ignore: invalid_use_of_internal_member
-          _native.appStartEnd = options.clock();
+          // We only assign the current time if it's not already set - this is useful in tests
+          _native.appStartEnd ??= options.clock();
+          final appStartEnd = _native.appStartEnd;
+          final nativeAppStart = await _native.fetchNativeAppStart();
+
+          if (nativeAppStart == null || appStartEnd == null) {
+            setAppStartInfo(null);
```

Comment:
maybe this is not needed, as we are going to use the `_didAddAppStartMeasurement` flag in the `apply` method
If for any reason this integration is called twice, we set the appStartInfo and right after that we remove it